gui/designwidget.cc

```diff
@@ -46,15 +46,29 @@ void TreeView::mouseMoveEvent(QMouseEvent *event)
 
 void TreeView::leaveEvent(QEvent *event) { Q_EMIT hoverIndexChanged(QModelIndex()); }
 
-DesignWidget::DesignWidget(QWidget *parent) : QWidget(parent), ctx(nullptr), selectionModel(nullptr)
+DesignWidget::DesignWidget(QWidget *parent) : QWidget(parent), ctx(nullptr)
 {
+    tabWidget = new QTabWidget();
+
     // Add tree view
-    treeView = new TreeView();
-    treeModel = new TreeModel::Model();
-    treeView->setModel(treeModel);
-    treeView->setContextMenuPolicy(Qt::CustomContextMenu);
-    treeView->setSelectionMode(QAbstractItemView::ExtendedSelection);
-    treeView->viewport()->setMouseTracking(true);
+    for(int i=0;i<6;i++)
+    {
+        treeView[i] = new TreeView();

```

Comment:
Instead of managing *treeView* by hand I'd rather have a vector of unique_ptrs here - that's less code and fewer mistakes. TreeView instances could be created with *std::make_unique<TreeView>()*;

Reply:
It looks like the same applies to *treeModel*.

Reply:
I might be wrong - I'm not sure how Qt manages these objects…